crypto/x509/t_x509.c

```diff
@@ -94,11 +94,11 @@ int X509_print_ex(BIO *bp, X509 *x, unsigned long nmflags,
         }
         if (l != -1) {
             if (bs->type == V_ASN1_NEG_INTEGER) {
-                l = -l;
+                l = -(unsigned long)l;
```

Comment:
The problem here is of course when it contains LONG_MIN. Maybe it would be better to have an unsigned long variable and then do `ul = 0  - (unsigned long)l;`? (The `0 - ` would fix the VC warning)